File: DSA/Vridhi_Jain_590014870/Day11_Ques1.c
#include <stdio.h>

int main() {
    char str[1000];
    char clean[1000];
    int i, j = 0, left, right;
    
    printf("Enter a string: ");
    scanf("%[^\n]", str);
    
    for (i = 0; str[i] != '\0'; i++) {
        if ((str[i] >= 'a' && str[i] <= 'z') || (str[i] >= '0' && str[i] <= '9')) {
            clean[j++] = str[i];
        } else if (str[i] >= 'A' && str[i] <= 'Z') {
            clean[j++] = str[i] + 32;
        }
    }
    
    left = 0;
    right = j - 1;
    
    while (left < right) {
        if (clean[left] != clean[right]) {
            printf("Not a palindrome\n");
            return 0;
        }
        left++;
        right--;
    }
    
    printf("Palindrome\n");
    return 0;
}

File: DSA/HarshitJain_500118968/Day11Ques1.java
import java.util.Scanner;

public class PalindromeIgnoreSpacesPunct {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String input = sc.nextLine();

        // Normalize: keep only alphanumeric, ignore case
        StringBuilder sb = new StringBuilder();
        for (char c : input.toCharArray()) {
            if (Character.isLetterOrDigit(c)) {
                sb.append(Character.toLowerCase(c));
            }
        }
        String filtered = sb.toString();

        // Check palindrome
        int left = 0, right = filtered.length() - 1;
        boolean isPalindrome = true;
        while (left < right) {
            if (filtered.charAt(left) != filtered.charAt(right)) {
                isPalindrome = false;
                break;
            }
            left++;
            right--;
        }
        System.out.println(isPalindrome);
    }
}